### ISSUE TITLE
Loss becomes infinity after some iterations

### ISSUE THREAD

**Issue by shivin9** [ashitani/darknet_mnist#2]
```mnist_lenet
1
layer     filters    size              input                output
    0 dilation: Using default '1'
conv     32  5 x 5 / 1 O 1   28 x  28 x   3   ->    28 x  28 x  32  0.004 BFLOPs
    1 max          2 x 2 / 2    28 x  28 x  32   ->    14 x  14 x  32
    2 dilation: Using default '1'
conv     64  5 x 5 / 1 O 1   14 x  14 x  32   ->    14 x  14 x  64  0.020 BFLOPs
    3 max          2 x 2 / 2    14 x  14 x  64   ->     7 x   7 x  64
    4 connected                            3136  ->  1024
    5 dropout       p = 0.50               1024  ->  1024
    6 connected                            1024  ->    10
    7 softmax                                          10
    8 cost                                             10
Learning Rate: 0.01, Momentum: 0.9, Decay: 5e-05
60000
28 28
Loaded: 0.008963 seconds
1, 0.002: 1.707400, 1.707400 avg, 0.009920 rate, 0.268078 seconds, 100 images
Loaded: 0.000051 seconds
2, 0.003: 1.633262, 1.699986 avg, 0.009841 rate, 0.091308 seconds, 200 images
Loaded: 0.000055 seconds
3, 0.005: 1.699964, 1.699984 avg, 0.009762 rate, 0.095921 seconds, 300 images
Loaded: 0.000073 seconds
4, 0.007: 1.588109, 1.688796 avg, 0.009684 rate, 0.097084 seconds, 400 images
Loaded: 0.000061 seconds
5, 0.008: 1.624906, 1.682407 avg, 0.009606 rate, 0.088261 seconds, 500 images
Loaded: 0.000050 seconds
6, 0.010: 1.667651, 1.680932 avg, 0.009529 rate, 0.091198 seconds, 600 images
Loaded: 0.000066 seconds
7, 0.012: 2.080166, 1.720855 avg, 0.009452 rate, 0.085911 seconds, 700 images
Loaded: 0.000058 seconds
8, 0.013: 3.659655, 1.914735 avg, 0.009375 rate, 0.094767 seconds, 800 images
Loaded: 0.000052 seconds
9, 0.015: inf, inf avg, 0.009299 rate, 0.089716 seconds, 900 images
Loaded: 0.000043 seconds
10, 0.017: inf, inf avg, 0.009224 rate, 0.090972 seconds, 1000 images
Loaded: 0.000057 seconds
11, 0.018: inf, inf avg, 0.009149 rate, 0.088817 seconds, 1100 images
Loaded: 0.000057 seconds
12, 0.020: 2.668562, inf avg, 0.009074 rate, 0.089873 seconds, 1200 images
Loaded: 0.000055 seconds
13, 0.022: 1.857954, inf avg, 0.009000 rate, 0.089997 seconds, 1300 images
Loaded: 0.000063 seconds
14, 0.023: 1.925004, inf avg, 0.008926 rate, 0.089042 seconds, 1400 images
Loaded: 0.000053 seconds
15, 0.025: 1.892647, inf avg, 0.008853 rate, 0.087421 seconds, 1500 images
Loaded: 0.340260 seconds
16, 0.027: 1.676757, inf avg, 0.008780 rate, 0.041032 seconds, 1600 images
Loaded: 0.000052 seconds
17, 0.028: 1.763951, inf avg, 0.008708 rate, 0.086645 seconds, 1700 images
Loaded: 0.000052 seconds
18, 0.030: 1.572006, inf avg, 0.008636 rate, 0.091281 seconds, 1800 images
Loaded: 0.000052 seconds
19, 0.032: 1.622870, inf avg, 0.008564 rate, 0.093262 seconds, 1900 images
Loaded: 0.000060 seconds
20, 0.033: 1.601578, inf avg, 0.008493 rate, 0.090092 seconds, 2000 images
Loaded: 0.000054 seconds
```

Any thoughts why this is happening?

**Comment by ashitani** [comment]
Please check what happens if *cfg/mnist_lenet.cfg* is changed to the following.
(batch_normalization lines are added and the activation functions are changed from ReLU to leaky ReLU)

```
[net]
batch=100
subdivisions=1
height=28
width=28
channels=3
momentum=0.9
decay=0.00005
max_crop=28

learning_rate=0.01
policy=poly
power=4
max_batches=500

angle=1
hue=1
saturation=1
exposure=1
aspect=1

[convolutional]
batch_normalize=1
filters=32
size=5
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=5
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[connected]
output= 1024
activation=leaky

[dropout]
probability=.5

[connected]
output= 10
activation=linear

[softmax]
groups=1

[cost]
type=sse
```

**Comment by shivin9** [comment]
Ok I fixed it by modifying my dataset. I made an RGB version of MNIST and that solved the problem.
